src/context/note-context.js

```diff
@@ -9,8 +9,13 @@ const useNote = () => useContext(NoteContext);
 
 
 const NoteContextProvider = ({children}) => {
+    const [showForm , setShowForm] = useState(false);
+    const [isEditNoteForm , setIsEditNoteForm] = useState(false);
+    const [title , setTitle] = useState('');
+    const [desc , setDesc] = useState('');
 
     const reducer = (accu , action) => {
```

Comment:
If you have multiple reducers you can rename them to be more specific like notesEditReducer, sortReducer etc.